addons/im_livechat/static/src/views/discuss_channel_list/discuss_channel_list_view_controller.js

```diff
@@ -21,13 +21,10 @@ export class DiscussChannelListController extends ListController {
                 additionalContext: { active_id: record.resId },
             });
         }
-        let thread = this.store.Thread.get({
+        const thread = await this.store.Thread.getOrFetch({
             model: "discuss.channel",
             id: record.resId,
         });
-        if (!thread?.type) {
-            thread = await this.threadService.fetchChannel(record.resId);
-        }
         if (thread) {
```

Comment:
The correct fix is to match with `master`:

```js
let thread = this.store.Thread.get({
    model: "discuss.channel",
    id: record.resId,
});
if (!thread?.channel_type) {
    thread = await thread.fetchChannelInfo();
}
```

`getOrFetch()` may return the channel if it has at least a `channel` matching `id` and `model`. However, this doesn't necessarily mean the channel info has been fetched, so this could be `undefined`.

Reply:
Oh ok, I thought `getOrFetch` was a `get` if the channel was already (partially) inserted. I forgot this acts as "get, and for channels, insert and make sure to at least fetch info once".

IMO this function has too many concerns, but this is a topic for `master` ;)